src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,465 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+
+//
+// This captures information of how various key methods of building with MSBuild ran.
+//
+namespace Microsoft.Build.Eventing
+{
+    // /OnlyProviders=*MSBuild-Profiling
+    [EventSource(Name = "Microsoft-Build")]
+    internal sealed class MSBuildEventSource : EventSource
+    {
+        #region Singleton instance
```

Comment:
We don't usually add `#region` in new code (we often delete it in old code). No strong reason IMO, just generally out of style.

Reply:
Regions make the code more organized. I agree this one is unnecessary, but I'd prefer to keep the one around events if that's ok.